collector-bitmovin-player/src/main/java/com/bitmovin/analytics/bitmovin/player/BitmovinSdkAdAdapter.kt

```diff
@@ -76,35 +110,47 @@ class BitmovinSdkAdAdapter(val bitmovinPlayer: BitmovinPlayer, val adAnalytics:
     }
 
     private val onAdQuartileListener = OnAdQuartileListener {
-        adAnalytics.onAdQuartile(adQuartileFactory.FromPlayerAdQuartile(it.quartile))
+        try {
+            adAnalytics.onAdQuartile(adQuartileFactory.FromPlayerAdQuartile(it.quartile))
+        } catch (e: Exception) {
+            Log.d(TAG, "On Ad Quartile Listener ", e)
+        }
     }
 
     init {
-        bitmovinPlayer.addEventListener(onAdStartedListener)
-        bitmovinPlayer.addEventListener(onAdFinishedListener)
-        bitmovinPlayer.addEventListener(onAdBreakStartedListener)
-        bitmovinPlayer.addEventListener(onAdBreakFinishedListener)
-        bitmovinPlayer.addEventListener(onAdClickedListener)
-        bitmovinPlayer.addEventListener(onAdErrorListener)
-        bitmovinPlayer.addEventListener(onAdSkippedListener)
-        bitmovinPlayer.addEventListener(onAdManifestLoadedListener)
-        bitmovinPlayer.addEventListener(onPlayListener)
-        bitmovinPlayer.addEventListener(onPausedListener)
-        bitmovinPlayer.addEventListener(onAdQuartileListener)
+        try {
+            bitmovinPlayer.addEventListener(onAdStartedListener)
+            bitmovinPlayer.addEventListener(onAdFinishedListener)
+            bitmovinPlayer.addEventListener(onAdBreakStartedListener)
+            bitmovinPlayer.addEventListener(onAdBreakFinishedListener)
+            bitmovinPlayer.addEventListener(onAdClickedListener)
+            bitmovinPlayer.addEventListener(onAdErrorListener)
+            bitmovinPlayer.addEventListener(onAdSkippedListener)
+            bitmovinPlayer.addEventListener(onAdManifestLoadedListener)
+            bitmovinPlayer.addEventListener(onPlayListener)
+            bitmovinPlayer.addEventListener(onPausedListener)
+            bitmovinPlayer.addEventListener(onAdQuartileListener)
+        } catch (e: Exception) {
+            Log.d(TAG, "On Ad init", e)
+        }
     }
```

Comment:
Do you think it is necessary here?
What could possibly go wrong?
same for release

Reply:
Well we don't have any control of what is happening in `addEventListener` method. It's highly unlikely that something will go wrong here, but still I would prefer to be safe :)